src/model.py

```diff
@@ -1342,26 +1360,31 @@ def scope_relevance(self, batch, shared_output) -> tuple():
 
         self.scope_true = self.relu(
             (labels["expression"] + labels["polarity"] + labels["target"])
-        ).bool().float().to(torch.device(self.device))
+        ).bool().long().to(torch.device(self.device))
         # scope_true.shape = [batch, sequence]
 
         # shared_output.shape = [batch, embedding (768), sequence]
-        self.scope_logits = self.components["scope"]["linear"](shared_output.permute(0, 2, 1)).squeeze(-1)
+        self.scope_logits = self.components["scope"]["linear"](shared_output.permute(0, 2, 1))
 
         # scope_logits.shape = [batch, sequence, 1]
-        self.scope_loss_value = self.scope_loss(self.scope_logits, self.scope_true)
+        self.scope_loss_value = self.scope_loss(self.scope_logits.permute(0, 2, 1), self.scope_true)
         self.scope_loss_value.backward(retain_graph=True)
         self.scope_optimizer.step()
 
         # TODO configurable guided start/warm_up?
         gold_influence = self.get_prob(self.current_epoch, self.find("warm_up_constant", default=5))
-        self.scope_output = (gold_influence*self.scope_true + (1-gold_influence)*self.scope_logits).detach()
+        try:  # FIXME remove this try/catch
+            self.scope_output = (
+                gold_influence*self.scope_true + (1-gold_influence)*self.scope_logits.argmax(-1).squeeze(-1)
+            ).detach()
```

Comment:
Scope also recieves a golden transmission, to guide model into predicting correct scope at begging of training